.github/workflows/test-go.yml

```diff
@@ -28,7 +28,7 @@ jobs:
     steps:
       - uses: actions/setup-go@v2
```

Comment:
Note that SDK is not used as a lib so nobody should be consuming it by importing.
Therefore that shows OK we bump its go. mod as well to 1.18 ( maybe in a follow up if you wish ) 
We need also clarifies on the changelog bump for go 1.18

Reply:
I think the go 1.18 bump for the OSDK go.mod can be a follow up.

I can add more details on the changelog about the go 1.18 bump.

Reply:
Actually found we needed to do the bump with this PR.